File: src/main/java/fyordo/lifeagragator/med/base/filters/UserIdFilter.java
package fyordo.lifeagragator.med.base.filters;

import jakarta.servlet.FilterChain;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Component;
import org.springframework.web.filter.OncePerRequestFilter;

import java.io.IOException;

@Component
public class UserIdFilter extends OncePerRequestFilter {

    @Override
    protected void doFilterInternal(
            @NonNull HttpServletRequest request,
            @NonNull HttpServletResponse response,
            @NonNull FilterChain filterChain
    ) throws ServletException, IOException {
        String userId = request.getHeader("X-User-Id");
        if (userId != null && !userId.isEmpty()) {
            request.getSession().setAttribute("userId", userId);
        }

        filterChain.doFilter(request, response);
    }
}

File: src/main/kotlin/fyordo/lifeagragator/med/base/filters/UserIdFilter.kt
package fyordo.lifeagragator.med.base.filters

import jakarta.servlet.FilterChain
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Component
import org.springframework.web.filter.OncePerRequestFilter

@Component
class UserIdFilter : OncePerRequestFilter() {
    override fun doFilterInternal(
        request: HttpServletRequest,
        response: HttpServletResponse,
        filterChain: FilterChain
    ) {
        val userId = request.getHeader("X-User-Id")
        if (userId != null && userId.isNotEmpty()) {
            request.session.setAttribute("userId", userId)
        }

        filterChain.doFilter(request, response)
    }

}